docs/container-set-template.md

```diff
@@ -116,3 +116,51 @@ Example B: Lopsided requests, e.g. `a -> b` where `a` is cheap and `b` is expens
 Can you see the problem here? `a` only has small requests, but the container set will use the  total of all requests. So it's as if you're using all that GPU for 10h. This will be expensive.
 
 Solution: do not use container set when you have lopsided requests.
+
+## `retryStrategy` usage
+
+> v3.3 and after
+
+You can set a `retryStrategy` to apply to all containers of a container set.
+
+This currently works differently from [retries](retries.md) for other template types.
+You can only set the `duration` between each retry and the total number of `retries`.
+
+See an example below:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: containerset-with-retrystrategy
+  annotations:
+    workflows.argoproj.io/description: |
+      This workflow creates a container set with a retryStrategy.
+spec:
+  entrypoint: containerset-retrystrategy-example
+  templates:
+    - name: containerset-retrystrategy-example
+      containerSet:
+        retryStrategy:
+          retries: "10" # if fails, retry at most ten times
+          duration: 30s # retry for at most 30s
+        containers:
+          # this container completes successfully, it won't retried.
```

Comment:
Yes according to my test there is no new node here， just retry in emissary.  This may be what confuses a lot of people, hopefully this PR will explain it.

The c3 retry twice, and c1 c2 no retry.
```
time="2024-03-29T06:40:25 UTC" level=info msg="capturing logs" argo=true
intentional failure
time="2024-03-29T06:40:25 UTC" level=debug msg="ignore signal child exited" argo=true
time="2024-03-29T06:40:26 UTC" level=info msg="capturing logs" argo=true
time="2024-03-29T06:40:26 UTC" level=debug msg="ignore signal urgent I/O condition" argo=true
intentional failure
time="2024-03-29T06:40:26 UTC" level=debug msg="ignore signal child exited" argo=true
time="2024-03-29T06:40:26 UTC" level=debug msg="forwarding signal terminated" argo=true
time="2024-03-29T06:40:27 UTC" level=info msg="sub-process exited" argo=true error="<nil>"
time="2024-03-29T06:40:27 UTC" level=info msg="not saving outputs - not main container" argo=true
Error: exit status 1
```

Reply:
The UI is:
![image](https://github.com/argoproj/argo-workflows/assets/72060326/31242b51-12f7-4056-a64a-b6db2a455c66)

Reply:
> The UI is:

Ohhh I got confused by the screenshots in the [issue](https://github.com/argoproj/argo-workflows/issues/11502) that have new nodes, but that's because they used a regular template-level `retryStrategy` in their first screenshot:

> 1. specifying a `retryStategy` with a `limit` of `2` at the template level (same level as `volumes`, `containerset`, `metadata`, etc.)

Thanks for clarifying this!

> Yes according to my test there is no new node here， just retry in emissary. 

It could be useful if Emissary could report the retried `node`s although that may technically require a new node type, since the `Container` itself does not get retried, the _process_ does.

> This may be what confuses a lot of people, hopefully this PR will explain it.

Yes, I think we should note this behavior explicitly in the docs. Let me think about how to best phrase that. Showing the logs as you did here may perhaps be the best way to visualize that (while also giving example logs for a user to check against)

Reply:
OK,Thanks!

Reply:
I add a note to describe the behavior, can you have a finnal check, thanks! @agilgur5

Reply:
Follow-up (hopefully final) review [below](https://github.com/argoproj/argo-workflows/pull/12809#pullrequestreview-1970042774)